main.py

```diff
@@ -25,27 +26,66 @@ def lambda_handler(event, context):
 
     s3_file_list = list_s3_files_matching_prefix(s3_client, prefix_string)
 
-    msg_queue = queue.Queue()
+    record_list = []
     for filename in s3_file_list:
-        record_list = extract_records_from_file(s3_client, filename)
-        [msg_queue.put(record) for record in record_list]
+        record_list.extend(extract_records_from_file(s3_client, filename))
 
+    external_validation_results = perform_external_validations(record_list)
+    internal_validation_results = perform_internal_validations(record_list)
 
-    # print(list(msg_queue.queue))
+    # if list_contains_no_errors(external_validation_results['Results']):
+    #     print("No validation errors detected.")
+    # else:
+    #     print("Validation failed, errors detected.")
+    #     if SEND_EMAIL_ALERTS:
+    #         print("Sending email alert.")
+    #         send_report("Validation failed, detected %d errors out of %d messages analyzed. ")
+    #     else:
+    #         print("Sending email alert.")
+    return
 
+### Validate message list using the validation library
+def perform_external_validations(msg_list):
+    msg_queue = queue.Queue()
+    [msg_queue.put(msg) for msg in msg_list]
     test_case = TestCase(CONFIG_FILE)
-    validation_results = test_case.validate_queue(msg_queue)
+    return test_case.validate_queue(msg_queue)
 
-    if list_contains_no_errors(validation_results['Results']):
-        print("No validation errors detected.")
-    else:
-        print("Validation failed, errors detected.")
-        if SEND_EMAIL_ALERTS:
-            print("Sending email alert.")
-            send_report("Validation failed, detected %d errors out of %d messages analyzed. ")
+### Validate things not covered by the validation library
+def perform_internal_validations(msg_list):
+    #sorted_list = sorted(msg_list, key=lambda msg: (json.loads(msg)['metadata']['logFileName'], json.loads(msg)['metadata']['serialId']['recordId']))
+    sorted_list = sorted(msg_list, key=lambda msg: (json.loads(msg)['metadata']['recordId'], json.loads(msg)['metadata']['serialId']['logFileName']))
```

Comment:
Looks like `serialId` is in the wrong place. Should be between `metadata` and `recordId`